python/pyspark/worker.py

```diff
@@ -39,6 +39,12 @@
 pickleSer = PickleSerializer()
 utf8_deserializer = UTF8Deserializer()
 
+try:
```

Comment:
Perhaps only do this if sentry is enabled?

Reply:
That's what happens here. Otherwise I would have to bring setuptools as runtime dependency to find out if pyspark was installed with sentry extra or not - that's not good idea.

Reply:
So this is more you're treating the package install as it being enabled. Generally in Spark we allow folks to configure new features separate from if the package is installed or not. But don't bother making changes until we hear what the other folks have to say anyways since I'm a little hesitant about this change.

Reply:
Sorry, I'm not familiar with pyspark packaging rules, so would very appreciate for any help here. 

My motivation to add it as extra was the same as the others are there: if, for instance, pandas is available, you can use pandas-related features of pyspark, but missing pandas should break pyspark. That's why pandas is defined in extras, not install requirements, right? Same is for raven.